csharp/src/Google.Protobuf/JsonTokenizer.cs

```diff
@@ -301,7 +307,7 @@ private void ValidateState(State validStates, string errorPrefix)
             /// </summary>
             private string ReadString()
             {
-                var value = new StringBuilder();
+                var builder = AcquireStringBuilder();
```

Comment:
We don't release this if we throw an exception. Presumably that's okay, it just means we won't reuse it - and given that an exception's being thrown, that's the least of our concerns. Maybe add a comment to that effect?